### PR TITLE
tools: move clang_tools under @envoy_dev

### PR DESCRIPTION
**Pull request by htuch**
Fixes regression in gen_compilation_database introduced in #8597, this causes unpleasant errors if a developer has not configured LLVM_CONFIG. The solution is to move developer tools into an `@envoy_api` like local external repository called `@envoy_dev`.

Risk level: Low
Testing: Manual, with LLVM_CONFIG set/unset.

Signed-off-by: Harvey Tuch <htuch@google.com>

### CODE REVIEW

**Comment by lizan** [comment]
This seems naive, and I believe it still break command like `bazel build //...` without LLVM_CONFIG? I'm thinking of splitting whole clang tools in a different bazel repository (like we do for `@envoy_api`), and only setup that repo when LLVM_CONFIG is set. WDYT?

**Comment by htuch** [comment]
The idea is to unblock folks affected by the regression, I agree it is naive. Sure, I'm happy to use a distinct repository like `@envoy_api`, but I probably don't have cycles today to work on this, likely early next week.

**Comment by htuch** [comment]
@lizan turns out I had a few cycles in between compiles, PTAL.

File: WORKSPACE [hunk 1]
```diff
@@ -15,3 +15,7 @@ envoy_dependencies()
 load("//bazel:dependency_imports.bzl", "envoy_dependency_imports")
 
 envoy_dependency_imports()
+
+load("//bazel:dev_binding.bzl", "envoy_dev_binding")
```
Comment (lizan) [inline WORKSPACE]: Can this two line rolled in to repositories.bzl?

File: bazel/dev_binding.bzl [hunk 2]
```diff
@@ -0,0 +1,39 @@
+def _default_envoy_dev_impl(ctx):
+    ctx.file("WORKSPACE", "")
```
Comment (lizan) [inline bazel/dev_binding.bzl]: add an if here to check LLVM_CONFIG exists?

File: WORKSPACE (hunk 1 above)
Comment (htuch) [reply WORKSPACE]: I think it makes sense to have them as an additional opt-in. Most folks working with WORKSPACE won't want them.

Comment (lizan) [reply WORKSPACE]: That's the point of having LLVM_CONFIG to control in repository impls?

Comment (htuch) [reply WORKSPACE]: As per @jpeach earlier point about not having too much magic, it's probably better to have an explicit statement "yes, I want dev tools" and then the conditioning.

Comment (lizan) [reply WORKSPACE]: A code change (incl. WORKSPACE) to say "yes, I want dev tools" seems too much hassle. Env vars isn't a magic.

Comment (htuch) [reply WORKSPACE]: Given the subjectivity of what is too much hassle, I'm happy to concede on this one, see update.

Comment (lizan) [reply WORKSPACE]: To be clear, "too much hassle" from Envoy consuming project perspective, (i.e. changing WORKSPACE in all of them to get dev tools). I expect these tools will be usable from there.

**Pull request merged**